src/wallet/redux/sagas/wallet.ts

```diff
@@ -63,15 +65,27 @@ export function* walletSaga(uid: string): IterableIterator<any> {
 
 
 function* handleSignatureRequest(wallet: ChannelWallet, requestId, positionData) {
-  // todo:
-  // - validate the transition
-  // - sign the position
-  // - store the position
+  // TODO: Validate transition
   const signedPosition = wallet.sign(positionData);
-
+  const state = decode(positionData);
+  yield storeLastSentState(wallet, state, signedPosition);
   yield put(actions.signatureSuccess(requestId, signedPosition));
 }
 
+function* storeLastSentState(wallet: ChannelWallet, state: State, signature: string) {
```

Comment:
I suppose you could just have one generator, `storeState`, and determine whether it was sent or received based on the player index.